.github/workflows/test.yml

```diff
@@ -44,3 +47,58 @@ jobs:
           source activate          
           pip install pytest
           pytest
+
+  paths-filter:
+    runs-on: ubuntu-latest
+    outputs:
+      tag-file-modified: ${{ steps.filter.outputs.tag-file }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Path Filter
+        id: filter
+        uses: dorny/paths-filter@v3
+        with:
+          filters: |
+            tag-file:
+              - 'ersilia/hub/content/metadata/tag.txt'
+
+      - name: Tag file modified
+        if: steps.filter.outputs.tag-file == 'true'
+        run: echo "Tag file modified"
+
+      - name: Tag file not modified
+        if: steps.filter.outputs.tag-file != 'true'
+        run: echo "Tag file not modified"
+
+  update-model-request:
+    runs-on: ubuntu-latest
+    needs: paths-filter
+    if: needs.paths-filter.outputs.tag-file-modified == 'true'
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install pyyaml
+
+      - name: Run update_model_request_template.py
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: python .github/scripts/update_model_request_template.py
+
+      - name: Commit and push changes
+        run: |
+          git config --local user.name "GitHub Actions"
```

Comment:
You can simply use the following commit and push action as it's being used [here](https://github.com/ersilia-os/ersilia/blob/2a0ba90821f9226392df2bbc4dcfe1ab77306115/.github/workflows/approve-dispatch.yml#L118). Also notice the author name and email id being used, with the token as `github_token: ${{ secrets.GITHUB_TOKEN }}` like you are doing now.

Reply:
Thanks @DhanshreeA. I have tried this. See the error on the [workflow run](https://github.com/ersilia-os/ersilia/actions/runs/10473939899/job/29006984569)

Reply:
Oh okay, no worries, I can take over this PR. The GH token generated for your account does not have enough permissions, my bad.

Reply:
Oh! Ok, thanks